### PR TITLE
FEA: add RaCT model

### PR DESCRIPTION
**Pull request by Zephyr-29**

### CODE REVIEW

File: recbole/model/general_recommender/ract.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+import torch.nn.functional as F
+import numpy as np
+
+from recbole.model.abstract_recommender import GeneralRecommender
+from recbole.model.init import xavier_normal_initialization
+from recbole.utils import InputType
+
+
+class RaCT(GeneralRecommender):
+    r"""RaCT is a item-based model collaborative filtering model that simultaneously rank all items for user .
```
Comment (hyp1231) [inline recbole/model/general_recommender/ract.py:25]: Please use the expression in the original paper directly.

File: recbole/trainer/trainer.py [hunk 2]
```diff
@@ -783,6 +784,91 @@ def evaluate(self, eval_data, load_best_model=True, model_file=None, show_progre
...
+
+    def fit(self, train_data, valid_data=None, verbose=True, saved=True, show_progress=False, callback_fn=None):
+        if self.model.train_stage == 'actor_pretrain':
+            return self.pretrain(train_data, verbose, show_progress)
+        elif self.model.train_stage == "critic_pretrain":
+            return self.pretrain(train_data, verbose, show_progress)
+        elif self.model.train_stage == 'finetune':
+            return super().fit(train_data, valid_data, verbose, saved, show_progress, callback_fn)
+        else:
+            raise ValueError("Please make sure that the 'train_stage' is 'pretrain' or 'finetune' ")
```
Comment (hyp1231) [inline recbole/trainer/trainer.py:871]: Two blank lines between classes.

```suggestion


            raise ValueError("Please make sure that the 'train_stage' is 'pretrain' or 'finetune' ")
```

File: recbole/model/general_recommender/ract.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        mlp_modules.append(nn.BatchNorm1d(3))
+        mlp_modules.append(nn.Linear(3, self.critic_layers_1))
+        mlp_modules.append(nn.ReLU())
+        mlp_modules.append(nn.Linear(self.critic_layers_1, self.critic_layers_2))
+        mlp_modules.append(nn.ReLU())
+        mlp_modules.append(nn.Linear(self.critic_layers_2, self.critic_layers_3))
+        mlp_modules.append(nn.ReLU())
+        mlp_modules.append(nn.Linear(self.critic_layers_3, 1))
+        mlp_modules.append(nn.Sigmoid())
+        return nn.Sequential(*mlp_modules)
```
Comment (hyp1231) [inline recbole/model/general_recommender/ract.py:184]: Maybe `critic_layers_1/2/3` can be summarized into one single arg (e.g. [100, 100, 10]).

Then we can use a loop to construct critic layers.

**Review by hyp1231 — CHANGES_REQUESTED** [review]
(*To be discussed*) `Trainer._build_optimizer` may need to be refactored.

**Review by 2017pxy — APPROVED** [review]

**Review by 2017pxy — APPROVED** [review]

**Pull request merged**